src/pages/index.js

```diff
@@ -191,49 +195,57 @@ function handleCardFormSubmit(inputValues) {
     })
     .catch((error) => {
       console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      addCardModal.renderSaving(false);
     });
-  addCardModal.renderSaving(false);
 }
 
 function handleDeleteClick(card) {
   deleteModalPopup.open();
   deleteModalPopup.setSubmitHandler(() => {
-    function deleteRequest() {
-      return api
-        .deleteCard(card._id)
-        .then(() => {
-          card.handleDelete();
-          console.log("Deleting card:", card);
-        })
-        .catch(console.error);
-    }
-    handleDeleteSubmit(deleteRequest);
+    deleteModalPopup.renderLoading(true, "Deleting...");
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.handleDelete();
+        console.log("Deleting card:", card);
+        deleteModalPopup.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        deleteModalPopup.renderLoading(false, "");
+      });
   });
 }
 
-function handleDeleteSubmit(deleteRequest) {
-  deleteModalPopup.renderLoading(true);
-  deleteRequest()
-    .then(() => {
-      deleteModalPopup.close();
-    })
-    .catch(console.error);
+// function handleDeleteSubmit(deleteRequest) {
+//   deleteModalPopup.renderLoading(true);
+//   deleteRequest()
+//     .then(() => {
+//       deleteModalPopup.close();
+//     })
+//     .catch(console.error);
 
-  deleteModalPopup.renderLoading(false);
-}
+//   deleteModalPopup.renderLoading(false);
+// }
 
 function handleLikeClick(card) {
   api
     .likeCard(card._id)
-    .then(card.toggleLikeBtn(), console.log(card._id + " has been liked"))
+    .then(() => {
+      card.toggleLikeBtn();
+      console.log(card._id + " has been liked");
+    })
     .catch(console.error);
 }
 
 function handleDislikeClick(card) {
-  api
-    .dislikeCard(card._id)
-    .then(card.toggleLikeBtn(), console.log(card._id + " has been disliked"))
-    .catch(console.error);
+  api.dislikeCard(card._id);
```

Comment:
* Likes don't work. See the errors https://disk.yandex.ru/i/uAHPOQCqIc5n-w 


Please, delete `;` from line 244


```js
function handleDislikeClick(card) {
  api
    .dislikeCard(card._id)
    .then(() => {
      card.toggleLikeBtn();
      console.log(card._id + " has been disliked");
    })
    .catch(console.error);
}


```